core/templates/domain/exploration/LostChangeObjectFactory.ts

```diff
@@ -63,24 +63,27 @@ export interface LostChangeBackendDict {
   'language_code'?: string;
 }
 
+// Properties are optional in 'LostChangeBackendDict' because all of them may
+// not be present in the dict and may change according to the cmd. Therefore,
+// they can be undefined.
 export class LostChange {
   cmd: string;
-  stateName: string;
-  newStateName: string;
-  oldStateName: string;
-  newValue: LostChangeValue;
-  oldValue: LostChangeValue;
-  propertyName: string;
-  contentId: string;
-  languageCode: string;
-  translationHTML: string;
+  stateName?: string;
+  newStateName?: string;
+  oldStateName?: string;
+  newValue?: LostChangeValue;
+  oldValue?: LostChangeValue;
+  propertyName?: string;
+  contentId?: string;
+  languageCode?: string;
+  translationHTML?: string;
```

Comment:
Should we specifically make the backend send nulls instead of sending partial dicts?

Reply:
See this https://github.com/oppia/oppia/blob/a4a3c2b7c673e928264919b595f51222ddd8e2c5/core/domain/exp_domain.py#L79-L85 .There are different cmds taking different parameters. Each of the property listed here is not mandatory to be present in other. I think current approach correctly fits in.

Reply:
Hmm, the ideal solution might be to have separate types for each type of cmd, but probably fine to keep it as is for now. Although TODO might make sense here.

Reply:
Done, added a TODO

Reply:
If you add a todo you need to add an issue and add it to the TODO (`# TODO(#99999): Explanation…`).

Reply:
Yeah done